storage/range.go

```diff
@@ -375,9 +375,13 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 		return r.newNotLeaderError()
 	} else if !held || expired {
 		// Otherwise, if not held by this replica or expired, request renewal.
-		if err := r.requestLeaderLease(timestamp); err != nil {
-			return err
+		err := r.requestLeaderLease(timestamp)
+		// Getting a LeaseRejectedError back means someone else got there
+		// first. Since it's committed by now, we can return a NotLeaderError.
+		if _, ok := err.(*proto.LeaseRejectedError); ok {
```

Comment:
if err is nil, then `ok = false`. If an `interface{}(*proto.LeaseRejectedError(nil))` got returned instead (which might be what you mean), we could check that `err.(*proto.LeaseRejectedError) != nil`, but a) that does not happen and b) even if it happened, we'd probably still do the same thing here.